migration_scripts/bulk_load/src/footprint.clj

```diff
@@ -309,3 +622,90 @@
              :tx-kinds
              :tx-recipients
              :tx-senders])))
+
+;; KV Store ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn offloaded-to-kv-store
+  "Simulate offloading blob data to the key-value store.
+
+  - Columns that have been marked to be moved into a key-value
+    store (according to `kv-columns`) have their sizes removed from
+    their corresponding table and gathered into a special `:kv-store`
+    entry in the sizes mapping.
+
+  - It's assumed that if a table has columns to be moved to a KV
+    store, all its TOAST-ed values belong to those columns, and will
+    have their size re-attributed to the `:kv-store`.
+
+  - It's also assumed that these columns will not impact indices and
+    therefore won't change index or primary key size, unless the entire
+    table can be moved to key-value store."
+  [tables]
+  (let [kv-size (atom 0)
+
+        ;; Indicate whether the column should be moved to the KV
+        ;; store (return `:mv`), removed from RDBMS (`:rm`),
+        ;; copied (`:cp`), or is not affected by offloading (`nil`).
```

Comment:
it seems like either `:mv` = `:rm` or `:mv` = `:cp`?

Reply:
See my comment above about what these four states correspond to (they are all distinct).